### PR TITLE
Add shellcheck to lint shell scripts

### PR DESCRIPTION
**Pull request by mlangbehn**
Now, all scripts in the tests directory are linted for best practices.

Signed-off-by: Matt Langbehn <matthew.langbehn@gmail.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,25 @@
 ---
+dist: trusty
+sudo: false
```
Comment (eggshell) [inline .travis.yml]: `sudo: false` is the default now. Not sure if explicitly defining it is wanted.

File: .travis.yml [hunk 2]
```diff
@@ -1,16 +1,25 @@
 ---
+dist: trusty
```
Comment (eggshell) [inline .travis.yml]: The [build environment page](https://docs.travis-ci.com/user/ci-environment/) is a little hazy on this, but iirc specifying distribution is reserved for virtual machine builds. I probably need to clarify this for myself, but it's a good conversation to work through.

File: tests/signed-off-by-test.sh [hunk 3]
```diff
@@ -4,10 +4,9 @@ success="true"
 
 for commit in $(git cherry master | cut -d " " -f 2)
 do
-    git show -s $commit | grep -q 'Signed-off-by:'
-    if [ $? -ne 0 ]; then
+    if ! git show -s "$commit" | grep -q 'Signed-off-by:'; then
```
Comment (eggshell) [inline tests/signed-off-by-test.sh:7]: We might want to reflect these changes over in hoist's testing. Also, I think this should be a separate commit in the interest of keeping a clean tree.

File: .travis.yml [hunk 4]
```diff
@@ -1,16 +1,25 @@
...
+
 cache:
   directories:
     - "$(npm config get prefix)/bin"
 
 language: python
 python: '2.7'
 
+addons:
+  apt:
```
Comment (eggshell) [inline .travis.yml:10]: According to [this](https://github.com/koalaman/shellcheck), it looks like we need the following in order to make sure shellcheck is in our package lists.
```
sources:
    - debian-sid
```

File: .travis.yml (hunk 2 above)
Comment (mlangbehn) [reply .travis.yml]: Precise doesn't have `shellcheck` in it's APT repos. About a month or so ago, I ran into this issue, and found that as a beta feature in Travis, you can require a Trusty-based Docker image, but must set `sudo: false` (hence, the next line).

...that said...it's still not able to find shellcheck, which I'm still looking into.

File: .travis.yml (hunk 1 above)
Comment (mlangbehn) [reply .travis.yml]: Not if you set `dist: trusty`.

File: tests/signed-off-by-test.sh (hunk 3 above)
Comment (mlangbehn) [reply tests/signed-off-by-test.sh:7]: I agree that this should be a separate commit, but was trying to keep this pull request to only one commit. Either way, this change is needed for shellcheck-test.sh to pass.

File: .travis.yml (hunk 4 above)
Comment (mlangbehn) [reply .travis.yml:10]: Huh. In my previous project, I only needed to set the distro to Trusty. I'll look into it.

File: tests/signed-off-by-test.sh (hunk 3 above)
Comment (eggshell) [reply tests/signed-off-by-test.sh:7]: I meant separate pull request rather than commit (my bad). Once that's opened we can merge that first.

File: .travis.yml (hunk 1 above)
Comment (eggshell) [reply .travis.yml]: Word, still a little hazy on that flowchart.

**Comment by eggshell** [comment]
Go ahead and rebase this on top of af673a9 and it should be good to go.

File: .travis.yml (hunk 4 above)
Comment (mlangbehn) [reply .travis.yml:10]: Fixed; see next two lines.

**Review by eggshell — APPROVED** [review]
